src/coreclr/Directory.Build.targets

```diff
@@ -26,7 +27,8 @@
           UseHardlinksIfPossible="false" />
   </Target>
 
-  <!-- Import targets here to have TargetPath and other macros defined. Limit to CoreLib. -->
-  <Import Condition="'$(MSBuildProjectName)' == 'System.Private.CoreLib' or '$(IsNativeAotProject)' == 'true'" Project="$(RepositoryEngineeringDir)illink.targets" />
+    <!-- Import targets here to have TargetPath and other macros defined. Limit to CoreLib. -->
+    <Import Condition="'$(MSBuildProjectName)' == 'System.Private.CoreLib' or '$(IsNativeAotProject)' == 'true'" Project="$(RepositoryEngineeringDir)illink.targets" />
+    <Import Project="$(RepositoryEngineeringDir)liveILLink.targets" />
```

Comment:
Apparently this causes crossgen2_publish.proj to reference the linker projects now. These projects flow some global properties between each other which causes binclashes, race conditions and in this case, restore errors because the project.assets.json file doesn't list a required RID.

see https://github.com/dotnet/runtime/pull/92826/files#r1341932080 for more details

Can we further fine tune this import to only apply to the projects that really need the live linker?

Reply:
Please look at a binlog and see if all the projects that now reference the linker project actually need it. I'm not sure about the crossgen2_publish case though.

Reply:
Crossgen2_publish needs the linker for platforms that don’t support NativeAOT.